app/models/post.rb

```diff
@@ -0,0 +1,22 @@
+class Post < ApplicationRecord
+  attribute :author_id, :integer
+  attribute :title, :string
+  attribute :text, :text
+  attribute :comments_count, :integer
+  attribute :likes_count, :integer
+
+  belongs_to :author, class_name: 'User', counter_cache: true
+  has_many :comments
+  has_many :likes
+
+  after_create :update_author_posts_count
+  after_destroy :update_author_posts_count
+
+  def update_author_posts_count
+    author.update(posts_count: author.posts.count)
+  end
+
```

Comment:
- [ **OPTIONAL** ] For good programming practice, I think it would be better if you use the `increment` method to update your counter as this gives several benefits such as `Encapsulation`, `Abstraction`, `Code Readability`, `Consistency`,... You can do it like this:
```ruby
 def update_post_counter
    author.increment!(:posts_counter)
  end
```
**I can suggest you use the same approach in your `Comment` and `Like` modals for the `#update_post_comments_count` and `#update_post_likes_count` methods**

Reply:
Thanks for your comments, however, I think it is a better option to use counter_cache, as you can see I am using it however I also created the methods required by the project requirement.
I don't think the option you suggest is to use the increment! is so appropriate since it doesn't take into account when an object is deleted. instead counter_cache handles everything automatically. without writing a lot of code.